packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -0,0 +1,372 @@
+import { type Fixture, fixture } from '../../utilities/tests/fixture';
+import {
+    TableRecord,
+    TableRowSelectionEventDetail,
+    TableRowSelectionMode
+} from '../types';
+import { TablePageObject } from './table.pageobject';
+
+interface SimpleTableRecord extends TableRecord {
+    stringData: string;
+}
+
+const simpleTableData = [
+    {
+        stringData: '0',
+        stringData2: 'a'
+    },
+    {
+        stringData: '1',
+        stringData2: 'b'
+    },
+    {
+        stringData: '2',
+        stringData2: 'c'
+    }
+] as const;
+
+// prettier-ignore
+async function setup(): Promise<Fixture<Table<SimpleTableRecord>>> {
+    return fixture<Table<SimpleTableRecord>>(
+        html`<nimble-table>
+            <nimble-table-column-text field-name="stringData">stringData</nimble-table-column-text>
+        </nimble-table>`
+    );
+}
+
+describe('Table row selection', () => {
+    let element: Table<SimpleTableRecord>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TablePageObject<SimpleTableRecord>;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await connect();
+        element.setData(simpleTableData);
+        await waitForUpdatesAsync();
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('defaults to selection mode of "none"', () => {
+        expect(element.selectionMode).toBe(TableRowSelectionMode.none);
+    });
+
+    it('configuration is valid when selection mode is "none" and an id field name is set', async () => {
+        element.selectionMode = TableRowSelectionMode.none;
+        element.idFieldName = 'stringData';
+        await waitForUpdatesAsync();
+
+        expect(element.checkValidity()).toBeTrue();
+    });
+
+    it('configuration is valid when selection mode is "none" and an id field name is not set', async () => {
+        element.selectionMode = TableRowSelectionMode.none;
+        element.idFieldName = undefined;
+        await waitForUpdatesAsync();
+
+        expect(element.checkValidity()).toBeTrue();
+    });
+
+    it('configuration is valid when selection mode is "single" and an id field name is set', async () => {
+        element.selectionMode = TableRowSelectionMode.single;
+        element.idFieldName = 'stringData';
+        await waitForUpdatesAsync();
+
+        expect(element.checkValidity()).toBeTrue();
+    });
+
+    it('configuration is invalid when selection mode is "single" and an id field name is not set', async () => {
+        element.selectionMode = TableRowSelectionMode.single;
+        element.idFieldName = undefined;
+        await waitForUpdatesAsync();
+
+        expect(element.checkValidity()).toBeFalse();
+        expect(element.validity.idFieldNameNotConfigured).toBeTrue();
+    });
+
+    it('is cleared when selection mode changes', async () => {
```

Comment:
It could be good to assert that the selection change event doesn't fire when selection changes due to the selection mode changing.